CMSIS RTOS portability samples give build warning for `numaker_m2l31ki`

**Describe the bug**
CMSIS RTOS portability samples are throwing a build warning when targeting the `numaker_m2l31ki`.

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using? numaker_m2l31ki
 
**To Reproduce**
Steps to reproduce the behavior:
1. west build -p always -b numaker_m2l31ki samples/subsys/portability/cmsis_rtos_v2/timer_synchronization/
2. See warning

**Expected behavior**
Build should complete without warnings

**Impact**
Blocking CI on #73886

**Logs and console output**
```
/home/danieldegrasse/zephyr/zephyrproject/zephyr/subsys/portability/cmsis_rtos_v2/wrapper.h:13: warning: "TRUE" redefined
   13 | #define TRUE    1
      |
In file included from /home/danieldegrasse/zephyr/zephyrproject/modules/hal/nuvoton/m2l31x/Devices/M2L31/Include/NuMicro.h:13,
                 from /home/danieldegrasse/zephyr/zephyrproject/zephyr/soc/nuvoton/numaker/m2l31x/./soc.h:11,
                 from /home/danieldegrasse/zephyr/zephyrproject/zephyr/modules/cmsis/./cmsis_core_m.h:24,
                 from /home/danieldegrasse/zephyr/zephyrproject/zephyr/modules/cmsis/./cmsis_core.h:10,
                 from /home/danieldegrasse/zephyr/zephyrproject/zephyr/include/zephyr/arch/arm/asm_inline_gcc.h:24,
                 from /home/danieldegrasse/zephyr/zephyrproject/zephyr/include/zephyr/arch/arm/asm_inline.h:18,
                 from /home/danieldegrasse/zephyr/zephyrproject/zephyr/include/zephyr/arch/arm/arch.h:33,
                 from /home/danieldegrasse/zephyr/zephyrproject/zephyr/include/zephyr/arch/cpu.h:19,
                 from /home/danieldegrasse/zephyr/zephyrproject/zephyr/include/zephyr/kernel_includes.h:36,
                 from /home/danieldegrasse/zephyr/zephyrproject/zephyr/include/zephyr/kernel.h:17,
                 from /home/danieldegrasse/zephyr/zephyrproject/zephyr/subsys/portability/cmsis_rtos_v2/thread.c:7:
/home/danieldegrasse/zephyr/zephyrproject/modules/hal/nuvoton/m2l31x/Devices/M2L31/Include/M2L31.h:725: note: this is the location of the previous definition
  725 | #define TRUE          (1UL)                ///< Boolean true, define to use in API parameters or return value
```

**Additional context**
This issue is fixed by https://github.com/zephyrproject-rtos/hal_nuvoton/pull/13

--- a/west.yml
+++ b/west.yml
@@ -193,7 +193,7 @@ manifest:
       groups:
         - hal
     - name: hal_nuvoton
-      revision: ab342e6915bf7bff2203a20985754f6dbdb5343d
+      revision: 466c3eed9c98453fb23953bf0e0427fea01924be
       path: modules/hal/nuvoton
       groups:
         - hal

